ogre/src/OgreLidarVisual.cc

```diff
@@ -149,6 +152,26 @@ void OgreLidarVisual::ClearVisualData()
 void OgreLidarVisual::SetPoints(const std::vector<double> &_points)
 {
   this->dataPtr->lidarPoints = _points;
+  this->dataPtr->pointColors.clear();
+  for (auto i = 0ul; i < this->dataPtr->lidarPoints.size(); i++)
```

Comment:
do you need an unsigned long? vector `size()` returns `size_t` which is an unsigned integer

this should work

```
 for (unsigned int i = 0u; i < this->dataPtr->lidarPoints.size(); ++i)
```